youtube_dl/extractor/gedidigital.py

```diff
@@ -0,0 +1,243 @@
+                            f.get('width'),
+                            f.get('height'),
+                            f['tbr']
+                        ]
+            return [None, None, None]
+
+        mobj = re.search(_PATTERN, mp4.get('mp4') or '')
+        if not mobj:
+            return None
+        pattern = mobj.group(1)
+
+        qualities = re.search(_PATTERN, mp4.get('manifest') or '')
+        if qualities:
+            qualities = qualities.group(2)
+            qualities = qualities.split(',') if qualities else ['.']
+            qualities = [i for i in qualities if i]
+        else:
+            qualities = [mobj.group(2)]
+
+        http_formats = []
+        for q in qualities:
+            w, h, t = get_format_info(q)
+            http_formats.append({
+                'url': mp4['mp4'].replace(pattern, 'rrtv-%s-' % q),
+                'width': w or _QUALITY[q][0],
+                'height': h or _QUALITY[q][1],
+                'tbr': t or int(q),
+                'protocol': 'https',
+                'format_id': 'https-%s' % q,
+            })
+        return http_formats
+
+    def _real_extract(self, url):
+        u = re.match(self._VALID_URL, url)
+        self.IE_NAME = u.group('iename') if u.group('iename') else 'gedi'
+        video_id = self._match_id(url)
+
+        webpage = self._download_webpage(url, video_id)
+        player_data = re.findall(
+            r'PlayerFactory\.setParam\(\'(?P<type>.+?)\',\s*\'(?P<name>.+?)\',\s*\'(?P<val>.+?)\'\);',
+            webpage)
+
+        formats = []
+        audio_fmts = []
+        title = None
+        thumb = None
+        mp4 = {}
+
+        for t, n, v in player_data:
+            if t == 'format':
+                if n == 'mp4':
+                    mp4.update({'mp4': v})
+                if n == 'video-hls-vod-ak':
+                    mp4.update({'manifest': v})
+                    formats.extend(self._extract_akamai_formats(
+                        v, video_id, {'http': 'media.gedidigital.it'}))
+                if n == 'audio-hls-vod':
+                    audio_fmts.extend(self._extract_m3u8_formats(
+                        v, video_id, 'm4a', m3u8_id='audio-hls', fatal=False))
+                if n == 'audio-rrtv-mp3':
+                    audio_fmts.append({
+                        'format_id': 'mp3',
+                        'url': v,
+                        'tbr': 128,
+                        'ext': 'mp3',
+                        'vcodec': 'none',
+                        'acodec': 'mp3',
+                    })
+            elif t == 'param':
+                if n == 'videotitle':
+                    title = v
+                if n in ['image_full_play', 'image_full', 'image']:
+                    thumb = v
+
+        title = self._og_search_title(webpage) if not title else title
+
+        # clean weird char
+        title = compat_str(title).encode('utf8', 'replace').replace(b'\xc3\x82', b'').decode('utf8', 'replace')
```

Comment:
do you have an example for this case?

Reply:
the last test has this problem.

https://video.espresso.repubblica.it/embed/tutti-i-video/01-ted-villa/14772/14870

if you open in the webrowser, check the source and look for 'videotitle' you will se the error

the title in the page is ok but the title present in the video player parameter is wrongly encoded by the website so it needs to be cleaned up.

Reply:
i think the use of `videotitle` is not reliable(it also has a problem with escaped double quotes), instead of this cleanup, other sources should be prefered.

Reply:
I think if I remember correctly why i went with `videotitle` was because not always the title was present in the page (some ebeds that I tested ad it missing or something) and instead `videotitle` was always present because is uno of the player parameter (but it need to be cleaned because is messed up a bit)

I'll collect some urls to do some testing and check better.

Reply:
done